src/extensionVariables.ts

```diff
@@ -7,29 +7,84 @@ import * as vscode from "vscode";
 import { IAzExtOutputChannel, IAzureUserInput, ITelemetryReporter } from "vscode-azureextensionui";
 import { LanguageClient } from "vscode-languageclient";
 import { isWebpack } from "./constants";
+import { assert } from "./fixed_assert";
 import { LanguageServerState } from "./languageclient/startArmLanguageServer";
 import { JsonOutlineProvider } from "./Treeview";
 
+/**
+ * Represents a scalar value that must be initialized before its getValue is called
+ */
+class InitializeBeforeUse<T> {
+    private _value: { value: T; initialized: true } | { initialized: false } = { initialized: false };
```

Comment:
>private _value: { value: T; initialized: true } | { initialized: false } = { initialized: false }; [](start = 4, length = 98)

Nit: I'd probably use similar technique for CachedValue, or make this class more similar to CachedValue for consistency :) It's interesting the stuff TS lets you do.

Reply:
I discovered this trick more recently. :-)

Actually not happy with how I'm having to do this class, but it's partly due to a bug in vscode.